UserAvatar/src/main/java/com/infomaniak/core/useravatar/exposed/UserAvatar.kt

```diff
@@ -0,0 +1,108 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.infomaniak.core.useravatar.exposed
+
+import android.content.res.Configuration
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.background
+import androidx.compose.foundation.border
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.sizeIn
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material3.Surface
+import androidx.compose.runtime.*
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.toArgb
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import coil3.compose.AsyncImage
+import coil3.request.ImageRequest
+import coil3.request.crossfade
+import com.infomaniak.core.useravatar.AvatarData
+import com.infomaniak.core.useravatar.AvatarDisplayState
+import com.infomaniak.core.useravatar.component.DefaultIconAvatar
+import com.infomaniak.core.useravatar.component.InitialsTextAvatar
+
+@Composable
+fun UserAvatar(modifier: Modifier = Modifier, avatarData: AvatarData, border: BorderStroke? = null) {
+
+    var isAvatarError by rememberSaveable(avatarData.uri) { mutableStateOf(false) }
+    val avatarDisplayState by rememberSaveable(avatarData, isAvatarError) {
+        val avatarState = when {
+            avatarData.uri.isNotBlank() -> AvatarDisplayState.Avatar
+            avatarData.userInitials.isNotBlank() -> AvatarDisplayState.Initials
+            else -> AvatarDisplayState.UnknownUser
+        }
+        mutableStateOf(avatarState)
+    }
+
+    val minAvatarSize = 32.dp
+    val context = LocalContext.current
+    val imageRequest = remember(avatarData.uri, context) {
+        ImageRequest.Builder(context)
+            .data(avatarData.uri)
+            .crossfade(true)
+            .build()
+    }
+
+    Box(
+        contentAlignment = Alignment.Center,
+        modifier = modifier
+            .sizeIn(minWidth = minAvatarSize, minHeight = minAvatarSize)
+            .clip(CircleShape)
+            .then(if (border == null) Modifier else Modifier.border(border = border, shape = CircleShape))
+            .then(if (avatarData.backgroundColor == null) Modifier else Modifier.background(color = Color(avatarData.backgroundColor))),
+    ) {
+        when {
+            isAvatarError && avatarDisplayState == AvatarDisplayState.Initials -> InitialsTextAvatar(avatarData)
+            isAvatarError -> DefaultIconAvatar(avatarData.iconColor)
```

Comment:
Is the DefaultIconAvatar associated with AvatarDisplayState.UnknownUser? If so, maybe name them the same way